cmd/cert-checker/main.go

```diff
@@ -238,10 +239,45 @@ var expectedExtensionContent = map[string][]byte{
 	"1.3.6.1.5.5.7.1.24": {0x30, 0x03, 0x02, 0x01, 0x05}, // Must staple feature
 }
 
+// checkValidations checks the database for matching authorizations that were
+// likely valid at the time the certificate was issued. Authorizations with
+// status = "deactivated" are counted for this, so long as their validatedAt
+// is before the issuance and expiration is after.
+func (c *certChecker) checkValidations(cert core.Certificate, dnsNames []string) error {
+	authzs, err := sa.SelectAuthzsMatchingIssuance(c.dbMap, cert.RegistrationID, cert.Issued, dnsNames)
+	if err != nil {
+		return fmt.Errorf("error checking authzs for certificate %s: %w", cert.Serial, err)
+	}
+
+	if len(authzs) == 0 {
+		return fmt.Errorf("no relevant authzs found valid at %s",
+			cert.Issued)
+	}
+
+	nameToAuthz := make(map[string]*corepb.Authorization)
+	for _, m := range authzs {
+		nameToAuthz[m.Identifier] = m
+	}
```

Comment:
It's not immediately clear to the reader that the set of authorizations returned by SelectAuthzsMatchingIssuance will have at most one authorization for each name, and that therefore this loop won't accidentally clobber one authz with another. But I guess that also doesn't really matter, because in theory all authzs returned from the SA should be equally-valid. Probably not worth changing anything here, just wanted to note that it took me a solid 5 minutes to convince myself that this was okay and other readers may do the same.